qemu/tests/qemu_guest_agent.py

```diff
@@ -1918,8 +1946,129 @@ def gagent_check_fsfreeze(self, test, params, env):
                 logging.warn("Finally failed to thaw guest fs,"
                              " detail: '%s'", detail)
             raise
+        # check after fsthaw
+        self._action_after_fsthaw(write_cmd_guest, write_cmd_timeout)
+
+    @error_context.context_aware
+    def gagent_check_fsfreeze(self, test, params, env):
+        """
+        Test guest agent commands "guest-fsfreeze-freeze"
+
+        Test steps:
+        1) Check the FS is thawed.
+        2) Freeze the FS.
+        3) Check the FS is frozen from both guest agent side and
+         guest os side.
+        4) Thaw the FS.
+        5) Check the FS is unfrozen from both guest agent side and
+         guest os side.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environmen.
+        """
+        self._fsfreeze()
```

Comment:
Is this necessary to have `_fsfreeze()` nested? Why not call `gagent_check_fsfreeze()` directly in `gagent_check_fsfreeze_list()`?

Reply:
I think this method is more clear.And the parameters are different.
def gagent_check_fsfreeze(self, test, params, env)
def _fsfreeze(self, fsfreeze_list=False, mountpoints=None,
                  check_mountpoints=None):

Reply:
@xiagao Well, not persuasive enough to me.

Reply:
But, I have no strong opinion on this, up to you.